auth/service_test.go

```diff
@@ -379,3 +442,185 @@ func TestDBAuthService_Authorize(t *testing.T) {
 		})
 	}
 }
+
+func TestDBAuthService_ListUsers(t *testing.T) {
+	cases := []struct {
+		name      string
+		userNames []string
+	}{
+		{
+			name:      "no_users",
+			userNames: []string{},
+		}, {
+			name:      "single_user",
+			userNames: []string{"foo"},
+		}, {
+			name:      "many_users",
+			userNames: []string{"foo", "bar", "baz", "quux"},
+		},
+	}
+	for _, testCase := range cases {
+		t.Run(testCase.name, func(t *testing.T) {
+			s := setupService(t)
+			for _, userName := range testCase.userNames {
+				if err := s.CreateUser(&model.User{DisplayName: userName}); err != nil {
+					t.Fatalf("CreateUser(%s): %s", userName, err)
+				}
+			}
+			gotList, _, err := s.ListUsers(&model.PaginationParams{Amount: -1})
+			if err != nil {
+				t.Fatalf("ListUsers: %s", err)
+			}
+			gotUsers := make([]string, 0, len(testCase.userNames))
+			for _, user := range gotList {
+				gotUsers = append(gotUsers, user.DisplayName)
+			}
+			sort.Strings(gotUsers)
```

Comment:
Is it a requirement that users are returned sorted? If so, removing this line would make the test stronger.